pyp2spec/pyp2conf.py

```diff
@@ -92,71 +100,70 @@ def pypi_version_or_macro(self, version):
             return "%{version}"
         return version
 
-    def license(self, compliant=False):
+    def filter_license_classifiers(self):
+        """Return the list of license classifiers defined for the package.
+
+        Filter out the parent categories `OSI-/DFSG Approved` which don't have any meaning.
+        """
+
+        return  [
+            c for c in self.package_data["info"]["classifiers"]
+            if (
+                c.startswith("License")
+                and c not in ("License :: OSI Approved", "License :: DFSG approved")
+            )
+        ]
+
+    def transform_to_spdx(self):
+        """Return SPDX identifiers and expression based on the found
+        package license metadata (classifiers or license keyword).
+        
+        If multiple identifiers are found, create an expression that's the safest option (with AND as joining operator).
+        Raise ValueError if no valid SPDX identifiers are found.
+        """
+
+        if (license_classifiers := self.filter_license_classifiers()):
+            identifiers = classifiers_to_spdx_identifiers(license_classifiers)
+            if identifiers:
+                expression = " AND ".join(identifiers)
+                return (identifiers, expression)
+
+        license_keyword = self.package_data["info"]["license"]
+        identifiers = license_keyword_to_spdx_identifiers(license_keyword)
+
+        # No more options to detect licenses left, hence explicit fail
+        if not identifiers:
+            raise NoLicenseDetectedError
```

Comment:
I personally dislike raising exception classes (rather than instances), but it's probably not wrong :confused: